### PR TITLE
refactor: rename StreamOperation -> StreamPermission

### PR DESCRIPTION
**Pull request by teogeb**
Renamed:
- `StreamOperation` -> `StreamPermission`
- remove `STREAM_` prefix of `StreamPermission` enum values
- `StreamListQuery.operation` -> `permission`

Not updated:
- `README.md` (is not up-to-date with API)
- commented-out code

### CODE REVIEW

File: packages/client/src/StreamRegistry.ts [hunk 1]
```diff
@@ -103,19 +103,19 @@ export class StreamRegistry implements Context {
         throw new NotFoundError('Stream: id=' + id)
     }
 
-    async hasPermission(streamId: string, userAddess: EthereumAddress, operation: StreamOperation) {
+    async hasPermission(streamId: string, userAddess: EthereumAddress, permission: StreamPermission) {
         return this.streamRegistryContractReadonly.hasPermission(streamId, userAddess,
-            StreamRegistry.streamOperationToSolidityType(operation))
+            StreamRegistry.StreamPermissionToSolidityType(permission))
```
Comment (timoxley) [inline packages/client/src/StreamRegistry.ts]: `StreamPermissionToSolidityType` should probably be `streamPermissionToSolidityType`, likely find/replace did this

File: packages/client/src/Stream.ts [hunk 2]
```diff
@@ -42,12 +42,12 @@ export interface StreamPermissions {
...
-    STREAM_DELETE = 'canDelete',
-    STREAM_PUBLISH = 'publishExpiration',
-    STREAM_SUBSCRIBE = 'subscribeExpiration',
-    STREAM_SHARE = 'share'
+export enum StreamPermission {
+    EDIT = 'edit',
+    DELETE = 'canDelete',
+    PUBLISH = 'publishExpiration',
+    SUBSCRIBE = 'subscribeExpiration',
+    SHARE = 'share'
```
Comment (timoxley) [inline packages/client/src/Stream.ts:50]: yeah this is nicer

**Review by timoxley — COMMENTED** [review]
Yeah this is definitely better naming and IIUC matches the terminology used in the smartcontract, however this will likely break everything outside this repo that interacts with permissions e.g. `core-frontend` (cc @juhah) and the client examples currently being developed (cc @fonty1 @bearni95) and maybe dataunions stuff (cc @jtakalai)

**Comment by hpihkala** [comment]
Note that the smart contract is still WIP and may change, especially on this kind of 'cleanup' stuff! Best to wait for that to be final?

File: packages/client/src/StreamRegistry.ts (hunk 1 above)
Comment (teogeb) [reply packages/client/src/StreamRegistry.ts]: Yeah, fixed

**Review by samt1803 — APPROVED** [review]

**Pull request merged**